pkgs/by-name/gl/globalprotect-openconnect_2/package.nix

```diff
@@ -0,0 +1,134 @@
+{
+  perl,
+  jq,
+  fetchFromGitHub,
+  lib,
+  openconnect,
+  libsoup,
+  webkitgtk_4_0,
+  pkg-config,
+  callPackage,
+  rustPlatform,
+  glib,
+  atk,
+  gdk-pixbuf,
+  pango,
+  cairo,
+  harfbuzz,
+  gtk3,
+  zlib,
+  makeDesktopItem,
+  includeUnfreeGui ? false,
+}:
+let
+  version = "2.3.7";
+  pname = "globalprotect-openconnect";
+  desktop = makeDesktopItem {
+    name = "GlobalProtect Openconnect VPN Client";
+    desktopName = "GlobalProtect Openconnect VPN Client";
+    comment = "A GUI for GlobalProtect VPN";
+    genericName = "GlobalProtect VPN Client";
+    exec = "gpclient launch-gui %u";
+    icon = "gpgui";
+    categories = [
+      "Network"
+      "Dialup"
+    ];
+    mimeTypes = [ "x-scheme-handler/globalprotectcallback" ];
+  };
```

Comment:
I would actually prefer we take the desktop entry from the upstream project.

https://github.com/NixOS/nixpkgs/blob/76612b17c0ce71689921ca12d9ffdc9c23ce40b2/pkgs/by-name/gp/gpclient/package.nix#L40

Reply:
If we don't have the gui, we don't need to unpack the projet, therefore, we wouldn't have the upstream desktop file which is necessary for the cli version I think for the callback to work properly.

Reply:
The packaging for gpclient shows that's not true. The desktop entry file is part of the git repo and not the proprietary release.